src/main/java/com/chess/backend/gamemodel/Move.java

```diff
@@ -35,43 +35,85 @@ public Move(Square from, Square to, Piece movedPiece, Piece takenPiece, Moves.ca
 //        }
     }
 
-    public Square getFrom()
-    {
+    public Square getFrom() {
         return this.from;
     }
 
-    public Square getTo()
-    {
+    public Square getTo() {
         return this.to;
     }
 
-    public Piece getMovedPiece()
-    {
+    public Piece getMovedPiece() {
         return this.movedPiece;
     }
 
-    public Piece getTakenPiece()
-    {
+    public Piece getTakenPiece() {
         return this.takenPiece;
     }
 
-    public boolean wasEnPassant()
-    {
+    public boolean wasEnPassant() {
         return this.wasEnPassant;
     }
 
-    public boolean wasPawnTwoFieldsMove()
-    {
+    public boolean wasPawnTwoFieldsMove() {
         return this.wasPawnTwoFieldsMove;
     }
 
-    public Moves.castling getCastlingMove()
-    {
+    public Moves.castling getCastlingMove() {
         return this.castlingMove;
     }
 
-    public Piece getPromotedPiece()
-    {
+    public Piece getPromotedPiece() {
         return this.promotedTo;
     }
+
+    @Override
+    public String toString() {
+        return "\nMove{" +
+                "\nfrom=" + from +
+                "\n, to=" + to +
+                "\n, movedPiece=" + movedPiece +
+                "\n, takenPiece=" + takenPiece +
+                "\n, promotedTo=" + promotedTo +
+                "\n, wasEnPassant=" + wasEnPassant +
+                "\n, castlingMove=" + castlingMove +
+                "\n, wasPawnTwoFieldsMove=" + wasPawnTwoFieldsMove +
+                '}';
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        Move move = (Move) o;
+        return wasEnPassant == move.wasEnPassant
+                && wasPawnTwoFieldsMove == move.wasPawnTwoFieldsMove
+                && getFrom().equals(move.getFrom())
+                && getTo().equals(move.getTo())
+                && getMovedPiece().equals(move.getMovedPiece())
+                && Objects.equals(getTakenPiece(), move.getTakenPiece())
+                && Objects.equals(promotedTo, move.promotedTo)
+                && getCastlingMove() == move.getCastlingMove();
+    }
+
+    @Override
+    public int hashCode() {
```

Comment:
what is the purpose of this method?

Reply:
This is needed for sets in java to work properly. Otherwise you will get duplicates of the same objects in a set which is illegal there.